packages/kit/svelte-kit.js

```diff
@@ -1,2 +1,11 @@
 #!/usr/bin/env node
-import './dist/cli.js';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+
+// in our own CI, and when deploying directly from this monorepo,
+// the `dist` directory will not exist yet
+if (fs.existsSync(fileURLToPath(new URL('./dist', import.meta.url)))) {
+	import('./dist/cli.js');
```

Comment:
Hmm, no import for the src cli? Could risk running sync from an outdated build

Reply:
No, the only time we use `src/cli.js`, we import that file explicitly. This is just to cover the case where `postinstall` is running _in this repo_ either in GitHub Actions or when we deploy the app to Cloudflare Pages or Vercel (since `postinstall` wasn't running in the test apps _anyway_, I added `svelte-kit sync` to their `check` scripts, and there's no other time we would need to run `postinstall` in the repo itself. I tried using `--ignore-scripts` instead, but then the site failed to deploy because of a dependency on `sharp` which needs `postinstall`)

Reply:
here's where we import `src/cli.js` explicitly: https://github.com/sveltejs/kit/blob/d7386d908ff8b48fd979b8e03470a37f9cd04985/packages/kit/test/cli.js#L5-L6

Reply:
So the only time we have to remember building it first is if we develop sync related hanges locally and want to run `node ../../../../../kit/src/svelte-kit.js sync` (or `pnpm svelte-kit sync`) in a test app to see what changes in the sync output we have to build kit first, otherwise it'll run the last built version (which may be from another branch even).

postinstall not running for workspace dependencies is a bit odd imho, i've asked about it in pnpm discord but didn't hear back.

Reply:
I'm fine with this small limitation to keep the code cleaner. The is_workspace and is_self code in the other version was cumbersome.